### PR TITLE
[kots] Allow multiple docker pull secrets

### PR DESCRIPTION
**Pull request by corneliusludmann**
## Description
This change allows using a different image registry for Gitpod than the registry that is used to pull the images in an air-gap setup.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/10298

## Release Notes

```release-note
[kots] Allow using a different image registry for Gitpod than the registry that is used to pull the images in an air-gap setup.
```

## Meta
/werft no-preview
/werft publish-to-kots

### CODE REVIEW

**Comment by roboquat** [comment]
@corneliusludmann: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: install/kots/manifests/gitpod-installer-job.yaml [hunk 1]
```diff
@@ -273,6 +272,26 @@ spec:
...
+              then
+                kubectl get secret "container-registry" -o=jsonpath="{.data['\.dockerconfigjson']}" | base64 --decode | yq -P - > registry-auth-external.yaml
+              else
+                yq eval 'select(.kind == "Secret" and .metadata.name == "builtin-registry-auth").data.".dockerconfigjson"' \
+                  "${GITPOD_OBJECTS}/templates/gitpod.yaml" | base64 --decode | yq -P - > registry-auth-builtin.yaml
+              fi
+              # merge all files together (https://mikefarah.gitbook.io/yq/operators/reduce#merge-all-yaml-files-together)
+              yq -o=json eval-all '. as $item ireduce ({}; . * $item )' registry-auth-*.yaml > merged-registry-auths.yaml
+              # create secret and update if exists (https://stackoverflow.com/a/45881259/1364435)
+              kubectl create secret generic merged-registry-auths \
```
Comment (corneliusludmann) [inline install/kots/manifests/gitpod-installer-job.yaml:290]: We could also create a secret manifest file and move it to the helm templates directory.

File: install/kots/manifests/gitpod-installer-job.yaml [hunk 2]
```diff
@@ -273,6 +272,26 @@ spec:
                 'del(select(.kind == "StatefulSet" and .metadata.name == "openvsx-proxy").status)' \
                 "${GITPOD_OBJECTS}/templates/gitpod.yaml"
 
+              # Merge different docker pull secrets into one secret
+              if [ '{{repl HasLocalRegistry }}' = "true" ];
+              then
+                kubectl get secret "{{repl ImagePullSecretName }}" -o=jsonpath="{.data['\.dockerconfigjson']}" | base64 --decode | yq -P - > registry-auth-airgap.yaml
```
Comment (mrzarquon) [inline install/kots/manifests/gitpod-installer-job.yaml:278]: `base64 --decode` should be `base64 -d` as --decode fails in busybox (fixed this in my test install)

**Pull request closed without merging**